### PR TITLE
Fix `_field_caps` serialization in order to support cross cluster search

### PR DESCRIPTION
**Pull request by s1monw**
Today the `_field_caps` API doesn't implement its request serialization
correctly since indices and indices options are not serialized at all.
This will likely break with all transport clients etc. and if this request
must be send across the network. This commit fixes this and adds correct
handling if we have only remote indices to prevent the inclusion of
all local indices.

### CODE REVIEW

**Review by nik9000 — APPROVED** [review]

**Review by javanna — APPROVED** [review]
LGTM should we also add a unit test for the field caps request serialization?

**Review by jimczi — APPROVED** [review]
LGTM
Can you also add the missing fields in FieldCapabilitiesRequestTests.

**Comment by spalger** [comment]
Works will with elastic/kibana#11114

**Comment by s1monw** [comment]
@javanna @jimczi pushed a new commit

**Review by jimczi — APPROVED** [review]
LGTM
Thanks for adding the tests

File: core/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesRequestTests.java [hunk 1]
```diff
@@ -33,10 +34,52 @@ private FieldCapabilitiesRequest randomRequest() {
...
+
+        // change fields
+        other.fields("foo", "bar");
+        assertNotEquals(request, other);
+        other.fields("bar");
+        assertEquals(request, request);
+
+        // change indices options
+        other.indicesOptions(IndicesOptions.strictExpand());
+        assertNotEquals(request, other);
```
Comment (javanna) [inline core/src/test/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesRequestTests.java]: maybe you can use EqualsHashCodeTestUtils#checkEqualsAndHashCode and save yourself some boilerplate here?

**Review by javanna — APPROVED** [review]
still LGTM thanks @s1monw left a tiny comment but it doesn't need another review and it can also be addressed later if applicable

**Pull request merged**